app/controllers/vulneruby_engine/nosql_injection_controller.rb

```diff
@@ -8,8 +8,23 @@ def index
       end
   
       def run
-        @result = SecretMongo.where(name: params[:name]).to_a
-        render('layouts/vulneruby_engine/nosql_injection/run')
+        begin
+          collection_name = "dummy_collections"
+          puts "NOSQLI PARAM: #{ params[:name]}"
+
+          client = Mongoid::Clients.default
+          db = client.database
+          collections = db.collection_names
+
+          client[collection_name, :capped => true, :size => 1024].create unless collections.include?(collection_name)
+          collection = db[collections[0].to_sym]
+          doc = {"name": SecureRandom.base64(5), "date": DateTime.now, "text": SecureRandom.base64(3) }
+          collection.insert_one(doc)
+          result = collection.find(name: params[:name]).first
```

Comment:
Should this be `@result` rather than `result`?

```suggestion
          @result = collection.find(name: params[:name]).first
```